authentication/pipeline/user.py

```diff
@@ -170,13 +171,44 @@ def validate_password(
     return {}
 
 
+# NOTE: This pipeline function should NOT be decorated with the @partial decorator. That could
+# present a potential security risk.
+def require_touchstone_login(
+        strategy, backend, flow=None, **kwargs
+):  # pylint: disable=unused-argument
+    """
+    If the user is attempting to log in via email and has authenticated via Touchstone/SAML, require
+    them to log in that way.
+
+    Args:
+        strategy (social_django.strategy.DjangoStrategy): the strategy used to authenticate
+        backend (social_core.backends.base.BaseAuth): the backend being used to authenticate
+        flow (str): the type of flow (login or register)
+    """
+    if backend.name != EmailAuth.name or flow != SocialAuthState.FLOW_LOGIN:
```

Comment:
Just realized this should probably be returning here if the `SAML_AUTH` feature is turned off, because there's no way for the user to fix the login at that point.

Reply:
Good point. Will do

Reply:
Should be all set